gems/sorbet-runtime/lib/types/private/methods/_methods.rb

```diff
@@ -345,13 +345,18 @@ def self.maybe_run_sig_block_for_method(method)
     run_sig_block_for_key(key) if has_sig_block_for_key(key)
   end
 
-  def self.run_sig_block_for_method(method)
-    run_sig_block_for_key(method_to_key(method))
+  def self.run_sig_block_for_method(method, sig_block)
+    run_sig_block_for_key(method_to_key(method), sig_block)
   end
 
-  private_class_method def self.run_sig_block_for_key(key)
+  private_class_method def self.run_sig_block_for_key(key, sig_block = nil)
     blk = @sig_wrappers[key]
-    if !blk
+    if sig_block
```

Comment:
I don't understand this code either. 
The code seems to always use `blk` below and even while it takes `sig_block` it checks that they are the same. Thus it seems equivalent to old code.
Am I missing something?